### PR TITLE
Refactor and make a few tweaks

### PR DESCRIPTION
**Pull request by Pandapip1**

### CODE REVIEW

**Comment by Pandapip1** [comment]
Feel free to squash this into other commits after merging this PR.

File: pkgs/by-name/wi/wivrn/package.nix [hunk 1]
```diff
@@ -36,58 +38,33 @@
...
     owner = "meumeu";
     repo = "wivrn";
-    rev = "v${finalAttrs.version}";
+    rev = "v${wivrnVersion}";
     hash = "sha256-RVRbL9hqy9pMKjvzwaP+9HGEfdpAhmlnnvqZsEGxlCw=";
   };
+  monadoVersion = builtins.head (builtins.elemAt (builtins.split
+    "monado\n +GIT_TAG +([A-Za-z0-9]+)"
+    (builtins.readFile (wivrnSrc + "/CMakeLists.txt"))
+  ) 1);
```
Comment (PassiveLemon) [inline pkgs/by-name/wi/wivrn/package.nix:54]: I'll have to let the other reviewers decide on this.

File: pkgs/by-name/wi/wivrn/package.nix [hunk 2]
```diff
@@ -127,18 +104,41 @@ stdenv.mkDerivation (finalAttrs: {
...
+  ] ++ monado.buildInputs;
 
   cmakeFlags = [
     (lib.cmakeBool "WIVRN_BUILD_CLIENT" false)
     (lib.cmakeBool "WIVRN_USE_VAAPI" true)
     (lib.cmakeBool "WIVRN_USE_X264" true)
     (lib.cmakeBool "WIVRN_USE_NVENC" cudaSupport)
+    (lib.cmakeBool "WIVRN_USE_SYSTEMD" true)
+    (lib.cmakeBool "WIVRN_USE_PIPEWIRE" true)
+    (lib.cmakeBool "WIVRN_USE_PULSEAUDIO" true)
```
Comment (PassiveLemon) [inline pkgs/by-name/wi/wivrn/package.nix:116]: Are you sure that both can be used?

File: pkgs/by-name/wi/wivrn/package.nix (hunk 1 above)
Comment (Pandapip1) [reply pkgs/by-name/wi/wivrn/package.nix:54]: This is necessary for the automatic update functionality to work.

File: pkgs/by-name/wi/wivrn/package.nix (hunk 2 above)
Comment (Pandapip1) [reply pkgs/by-name/wi/wivrn/package.nix:116]: Yup. This just builds support for it; if the audio backend isn't present, it just doesn't use it.

**Comment by PassiveLemon** [comment]
I appreciate the suggestions

**Pull request closed without merging**